### ISSUE TITLE
Implement support for ITrackingConsentFeature ?

### ISSUE THREAD

**Issue by jcemoller** [Microsoft/ApplicationInsights-aspnetcore#838]
As part of GDPR implementation, we are restricting the usage of cookies based on user consent. We properly set the new _ITrackingConsentFeature_ based on the user's choice. 

The _JavaScriptSnippet_ part of Application Insights does not seem to react properly to this. Also, _isCookieUseDisabled_ does not seem to be exposed in any configuration object.

Workaround right now is to implement something like this

```razor
@Html.Raw(JavaScriptSnippet.FullScript)
@if (!hasCookieConsent())
{
<script>
    window.appInsights.config.isCookieUseDisabled = true;
</script>
}
```

**Q:** Would it be possible to extend the default _JavaScriptSnippet_ implementation to react to _ITrackingConsentFeature_ and automatically set _isCookieUseDisabled_ based on user consent?

**Comment by Dmitry-Matveev** [comment]
Hello @jcemoller ,

Thanks for your suggestion. Currently the burden of obtaining consent and managing cookies use is on the app owner (e.g. with the workaround you provided above). It would make sense to ease that burden by reacting to well-known and established flags in our SDK code.

Unfortunately, the planning for the next phase is over and I cannot provide a solid date for us to take action on this new feature request. Marking it as an enhancement and moving into the Future milestone to triage for the later releases. 

@cijothomas, please take a look if you have a chance and if you think we can move it to the closer milestone.

**Comment by cijothomas** [comment]
Still planned for "Future" i.e no ETA. 
Please continue to upvote the issue.

**Comment by itanex** [comment]
I am trying to implement this work around functionality, obviously the return value of the function `hasCookieConsent()` is going to be a bool, what specifics do you check for in the function itself? Is there any of the AppInsights API that is interacted with, or is there just some custom processes that are being made?

**Comment by jcemoller** [comment]
@itanex yes I've got a custom solution built on top of [CookieBot](https://www.cookiebot.com/) but you could of course use any mechanism to detect if user have consented.
